packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -73,42 +78,82 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
+
+    getOptionPrice(item, selectedValues, bundleDefaultPrices, isDynamicPrice) {
         const { option_id } = item;
         let price = 0;
         let priceExclTax = 0;
+        let initialPrice = 0;
 
         selectedValues.forEach(({ id, quantity, value }) => {
             if (option_id === id) {
                 const { options } = item;
 
-                options.forEach(({ id: optionId, product }) => {
+                options.forEach(({
+                    id: optionId, product, price: optionPrice, price_type: priceType
+                }) => {
                     if (JSON.stringify(value) === JSON.stringify([optionId.toString()])) {
                         if (product === null) {
                             return;
                         }
 
-                        const {
-                            price_range: {
-                                minimum_price: {
-                                    final_price: {
-                                        value: itemPrice = 0
-                                    } = {},
-                                    final_price_excl_tax: {
-                                        value: itemPriceExclTax = 0
-                                    } = {}
-                                } = {}
-                            } = {}
-                        } = product;
-
-                        price += itemPrice * quantity;
-                        priceExclTax += itemPriceExclTax * quantity;
+                        if (isDynamicPrice) {
+                            const {
+                                optionPrice,
+                                optionPriceExclTax,
+                                optionInitialPrice
+                            } = this.getDynanicOptionPrice(product, quantity);
+
+                            price += optionPrice;
+                            priceExclTax += optionPriceExclTax;
+                            initialPrice += optionInitialPrice;
+                        } else if (priceType === PRICE_TYPE_FIXED) {
+                            initialPrice += optionPrice * quantity;
+                            price += optionPrice * quantity;
+                            priceExclTax += optionPrice * quantity;
+                        } else if (priceType === PRICE_TYPE_PERCENT) {
+                            const {
+                                base_price: { value: defaultPrice } = 0,
+                                base_final_price: { value: defaultFinalPrice } = 0,
+                                base_final_price_excl_tax: { value: defaultFinalPriceExclTax } = 0
+                            } = bundleDefaultPrices;
+
+                            initialPrice += (defaultPrice * optionPrice * quantity) / ONE_HUNDRED_PERCENT;
+                            price += (defaultFinalPrice * optionPrice * quantity) / ONE_HUNDRED_PERCENT;
+                            priceExclTax += (defaultFinalPriceExclTax * optionPrice * quantity) / ONE_HUNDRED_PERCENT;
```

Comment:
I don't get why do you multiply prices like `defaultPrice * optionPrice`.
Should there be a sum or is this a discount?
Please give explanatory comments or variable names.

Reply:
<img width="552" alt="Bag bundle | Scandipwa 2021-04-01 13-52-50" src="https://user-images.githubusercontent.com/79456428/113284319-21b23480-92f2-11eb-9c58-95980f2724d4.png">

The way it is calculated is not straightforward, I has to check how Luma does it:
<img width="640" alt="Bag bundle Scandipwa 2021-04-01 13-59-11" src="https://user-images.githubusercontent.com/79456428/113284654-9be2b900-92f2-11eb-8031-e8f93e1b8bc9.png">

Reply:
Got it. Thanks. But could you put a comment in code or rename the variable so we can implicitly know that this is percents?
Because even if we debug values, we will see integers, not floats. And this is confusing despite this is a default magento behavior.